components/net/http_loader.rs

```diff
@@ -2057,10 +2052,16 @@ async fn http_network_fetch(
 
     if let Some(ref sender) = devtools_sender {
         let sender = sender.lock().unwrap();
-        if let Some(m) = msg {
-            send_request_to_devtools(m, &sender);
+        if let Some(ChromeToDevtoolsControlMsg::NetworkEvent(_, NetworkEvent::HttpRequest(devtools_request))) = msg {
+            let update_msg = ChromeToDevtoolsControlMsg::NetworkEvent(
+                request.id.0.to_string(),
+                NetworkEvent::HttpRequestUpdate(devtools_request),
```

Comment:
Can we change the code in obtain_resoonse to return a NetworkEventUpdate instead of creating that message here?